bootstrap_cfn/fab_tasks.py

```diff
@@ -701,3 +802,118 @@ def cycle_instances(delay=None):
+                     "Error updating dns record: "
+                     "{}".format(e.message))
+
+    # get all public facing elbs
+    elb = get_one_public_elbs()
+    # helloworld.dsd.io
+    main_record_name = "{}.{}".format(elb, zone_name)
+    # helloworld-12345.dsd.io
+    stack_record_name = "{}-{}.{}".format(elb, tag_stack_id, zone_name)
+    # get the ELB value in stack_record_name's record
+    record_name = "{}-{}".format(elb, tag_stack_id)
+    record_object = r53_conn.get_full_record(zone_name, zone_id, record_name, 'A')
+    record_value = [record_object.alias_hosted_zone_id,
+                    record_object.alias_dns_name,
+                    record_object.alias_evaluate_target_health]
+    if record_value:
+        # point [helloworld.dsd.io] to [helloworld-12345.dsd.io]'s ELB
+        try:
+            r53_conn.update_dns_record(zone_id, main_record_name, 'A', record_value, is_alias=True)
+            logger.info("fab_tasks::set_active_stack: Successfully updated dns alias record")
+            print green("Active stack is switched to {}".format(tag_record))
+        except Exception as e:
+            logger.error("fab_tasks::set_stack_name:"
+                         "Error updating dns record: "
+                         "{}".format(e.message))
+        return True
+    else:
+        raise StackRecordNotFoundError(stack_record_name)
+
+
+@task
+def get_active_stack():
+    """
+    Returns stack id if active stack exists AND Alias record is set appropriately
+    """
+    try:
+        active_record = get_tag_record_name('active')
+        r53_conn = get_connection(R53)
+        zone_name = get_zone_name()
+        zone_id = get_zone_id()
+        active_stack_id = r53_conn.get_record(zone_name, zone_id, active_record, 'TXT')
+        elb = get_one_public_elbs()
+        dns_record_name = '{}-{}'.format(elb, active_stack_id)
+        dns_record_value = r53_conn.get_record(zone_name, zone_id, dns_record_name, 'A')
+        main_record_value = r53_conn.get_record(zone_name, zone_id, elb, 'A')
+    except:
+        print green("No active stack exists.")
+        return None
+    if active_stack_id and dns_record_value and dns_record_value == main_record_value:
+        logger.info("fab_tasks::get_active_stack: "
+                    "Active stack id is: {}".format(active_stack_id))
+        print green("Active stack id is: {}".format(active_stack_id))
+        return active_stack_id
+    else:
+        print green("No active stack exists.")
+        return None
+
+
+def get_all_elbs():
+    """
+    Returns all internet-facing elbs from cloudformation configuration
+    """
+    cfn_config = get_config()
+    elbs = [x.get('name') for x in cfn_config.data.get('elb', {}) if x.get('scheme') == 'internet-facing']
+    return elbs
+
+
+def get_one_public_elbs():
+    """
+    Returns the first elb if exists.
+    """
+    elbs = get_all_elbs()
+    if len(elbs) < 1:
+        raise PublicELBNotFoundError
```

Comment:
All I am saying is that there are two cases where we can create a stacks that don't fit this restriction you have imposed. 1) A stack without an ELB 2) a stack with a non internet facing ELB. I haven't tested it but at least the 2nd case is valid I believe. So maybe raising an exception is not suitable here.